### PR TITLE
Added method to determine best quality image for flickr ripper [#1324]

### PR DESCRIPTION
**Pull request by jbatesy**
# Category

This change is exactly one of the following (please change `[ ]` to `[x]`) to indicate which:
* [ ] a bug fix (Fix #...)
* [ ] a new Ripper
* [ ] a refactoring
* [ ] a style change/fix
* [x ] a new feature

# Description

Please add details about your change here.

This was requested in issue #1324 and was also mentioned in FlickrRipper.java as being on the todo list. The feature simply adds a method to determine the best quality/largest image available using the Flickr API. Best quality is determined by height * width. getURLsFromPage() originally had a primitive method of guessing the largest image but was not very effective and failed to always grab the best quality image.

# Testing

Required verification:
* [x ] I've verified that there are no regressions in `mvn test` (there are no new failures or errors).
only fails were preexisting
* [x ] I've verified that this change works as intended.
  * [x ] Downloads all relevant content.
  * [ x] Downloads content from multiple pages (as necessary or appropriate).
  * [x ] Saves content at reasonable file names (e.g. page titles or content IDs) to help easily browse downloaded content.
* [x ] I've verified that this change did not break existing functionality (especially in the Ripper I modified).

Optional but recommended:
* [ ] I've added a unit test to cover my change.

### CODE REVIEW

**Comment by jbatesy** [comment]
@cyian-1756 any issues with this pull request? its my first ever public commit so feedback is welcome :)

File: src/main/java/com/rarchives/ripme/ripper/rippers/FlickrRipper.java [hunk 1]
```diff
@@ -250,4 +241,22 @@ public Document getFirstPage() throws IOException {
...
+        TreeMap<Integer, String> imageURLMap = new TreeMap<>();
+
+        try {
+            JSONArray imageSizes = new JSONObject(Http.url(imageAPIURL).ignoreContentType().get().text()).getJSONObject("sizes").getJSONArray("size");
+            for (int i = 0; i < imageSizes.length(); i++) {
+                JSONObject imageInfo = imageSizes.getJSONObject(i);
+                imageURLMap.put(imageInfo.getInt("width") * imageInfo.getInt("height"), imageInfo.getString("source"));
+            }
+
+        } catch (org.json.JSONException ignore) {
```
Comment (cyian-1756) [inline src/main/java/com/rarchives/ripme/ripper/rippers/FlickrRipper.java]: Don't silently throwaway errors, it makes debugging a pain. Add a logging statement explaining what happened

**Comment by cyian-1756** [comment]
> any issues with this pull request?

Only issue I found was exception being silently ignored. Aside from that every looks fine

**Comment by jbatesy** [comment]
> Only issue I found was exception being silently ignored.

Just fixed the way I handled exceptions.

**Pull request merged**